array_family/flex_ext.py

```diff
@@ -469,6 +471,48 @@ def subsort(self, key0, key1, reverse=False):
             ref_tmp.sort(key1, reverse)
             self[min(val) : (max(val) + 1)] = ref_tmp
 
+    def match_by_hkle(
+        self, other: dials_array_family_flex_ext.reflection_table
+    ) -> Tuple[cctbx.array_family.flex.size_t, cctbx.array_family.flex.size_t,]:
+        """
+        Match reflections with another set of reflections by the h, k, l
+        and entering values. Uses pandas dataframe merge method to match
+        the columns: assumes the key h, k, l, e is unique which is false
+        if > 360 degree rotation.
+
+        :param other: reflection table to match against
+        :return: indices in self, indices in other for matches
+        """
+
+        hkl = self["miller_index"].as_vec3_double().parts()
+        h0 = hkl[0].iround().as_numpy_array()
+        k0 = hkl[1].iround().as_numpy_array()
+        l0 = hkl[2].iround().as_numpy_array()
+        e0 = self["entering"].as_numpy_array()
+        n0 = np.array(range(len(e0)))
+
+        x0 = {"h": h0, "k": k0, "l": l0, "e": e0, "n0": n0}
+
+        p0 = pd.DataFrame(data=x0, columns=["h", "k", "l", "e", "n0"])
+
+        hkl = other["miller_index"].as_vec3_double().parts()
+        h1 = hkl[0].iround().as_numpy_array()
+        k1 = hkl[1].iround().as_numpy_array()
+        l1 = hkl[2].iround().as_numpy_array()
+        e1 = other["entering"].as_numpy_array()
+        n1 = np.array(range(len(e1)))
+
+        x1 = {"h": h1, "k": k1, "l": l1, "e": e1, "n1": n1}
+
+        p1 = pd.DataFrame(data=x1, columns=["h", "k", "l", "e", "n1"])
```

Comment:
An alternative approach, using `flumpy` could be:
```
from dxtbx import flumpy
...
h, k, l = np.hsplit(flumpy.to_numpy(hkl.as_vec3_double()).astype(int), 3)
```

It looks like `flumpy` doesn't natively handle `flex.miller_index` types, but since this is something can control we should be able to easily add this, then we can skip the flex.miller_index -> flex.vec3_double() -> int dance.

Reply:
I like the idea of avoiding the dance. I like it so much: should we add that to `flumpy` and work into this PR _before_ consider merging?